sqlmesh/core/test/definition.py

```diff
@@ -549,4 +549,7 @@ def _normalize_dataframe(value: t.Any) -> t.Any:
     """Normalize data in a pandas dataframe so ruamel and sqlglot can deal with it."""
     if isinstance(value, np.ndarray):
         return [_normalize_dataframe(v) for v in value]
+    if isinstance(value, dict) and "key" in value and "value" in value:
+        # {'key1': 10, 'key2': 20, 'key3': 30} looks like {'key': ['key1', 'key2', 'key3'], 'value': [10, 20, 30]}
+        return dict(zip(value["key"], _normalize_dataframe(value["value"])))
```

Comment:
We may need to parameterize this dict normalization logic per dialect. This will depend on whether other engines follow this format for maps.